deploy/runtools/run_farm.py

```diff
@@ -361,34 +390,34 @@ def terminate_run_farm(self, terminatesomef1_16, terminatesomef1_4, terminatesom
         else:
             rootLogger.critical("Termination cancelled.")
 
-    def get_all_host_nodes(self):
+    def get_all_host_nodes(self) -> List[EC2Inst]:
         """ Get objects for all host nodes in the run farm that are bound to
         a real instance. """
-        allinsts = self.f1_16s + self.f1_2s + self.f1_4s + self.m4_16s
-        return [inst for inst in allinsts if inst.boto3_instance_object is not None]
+        allinsts: List[EC2Inst] = [*self.f1_16s, *self.f1_2s, *self.f1_4s, *self.m4_16s]
```

Comment:
This is hacky because lists of different types can't be added together. The type checker doesn't work in this scenario: 

```
class A(X)
class B(X)
class C(X)

all: List[X] = [A] + [B] + [C] # errors since A can't be added to B (doesn't resolve to X)
```